src/integrations.py

```diff
@@ -72,9 +85,14 @@ def __init__(self, charm: CharmBase):
         self._charm = charm
         self._bind_account: Optional[BindAccount] = None
 
-    def load_bind_account(self, user: str, group: str) -> None:
+    def load_bind_account(self, user: str, group: str, relation_id: int) -> None:
         database_config = DatabaseConfig.load(self._charm.database_requirer)
         self._bind_account = _create_bind_account(database_config.dsn, user, group)
+        if not self._bind_account.password:
```

Comment:
Have an "audacious" thought: is it OK if we just reset the bind password if the bind account exists? 

Just to keep it naively simple because re-integration normally does not happen frequently, and also the consumer charm is supposed to update its config whenever the provider ldap data is ready.

Reply:
I thought of this as well, I will give it a go

Reply:
`relation-changed` will be fired every time a unit joins a relation. This means that every time the remote app is scaled, we will issue multiple new revisions of the secret. 

I will leave it as it is for now.